codice/ddf
# DDF-2317 Set metacard.created and metacard.modified dates in Metacard Groomer Plugin to align to new taxonomy

#### What does this PR do?
To support the new DDF taxonomy, the Metadata Groomer pre-ingest plugin was updated to set the Metacard.CREATED and Metacard.MODIFIED fields upon ingest to the current date/time for the new ddf.catalog.data.types.Metacard class. Additionally, the Metacard.MODIFIED field is updated to the current date/time for an update operation. 

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining
@jrnorth
@brendan-hofmann
@coyotesqrl

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire

#### How should this be tested?
Build and run DDF.
Verify that after ingest, the metacard reflects the current date/time for both `metacard.created` and `metacard.modified` date/time values.

**_Note: The transformers in their current state do not support the new taxonomy. Please use the following as a definitions.json file to dynamically apply these attributes across all metacard types for testing._**

```
{
    "attributeTypes": {
        "metacard.created": {
            "type": "DATE_TYPE",
            "stored": true,
            "indexed": true,
            "tokenized": false,
            "multivalued": false
        },
        "metacard.modified": {
            "type": "DATE_TYPE",
            "stored": true,
            "indexed": true,
            "tokenized": false,
            "multivalued": false
        }
    },
    "inject": [
        {
            "attribute": "metacard.created"
        },
        {
            "attribute": "metacard.modified"
        }
    ]
}  
```

#### Any background context you want to provide?
This capability supports the Expiration Date Plugin by setting the `metacard.created` attribute, which is used to calculate the expiration date for the metacard. 

#### What are the relevant tickets?
_Metacard created and modified dates should be set during ingest to reflect the new taxonomy mappings_
https://codice.atlassian.net/browse/DDF-2317

_As an administrator, I want data purged based on its age so that I can control the size of the catalog_
https://codice.atlassian.net/browse/DDF-2274

#### Checklist:
